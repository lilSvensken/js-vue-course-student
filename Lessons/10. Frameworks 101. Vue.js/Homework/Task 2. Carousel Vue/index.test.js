/**
 * @jest-environment jsdom
 */

const { mount } = require("@vue/test-utils");
const Carousel = require("./Carousel.vue");

describe("10.h.2 Carousel", () => {
    const wrapper = mount(Carousel.default, {
        data() {
            return {
                images: [
                    '/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg',
                    '/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg',
                    '/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg',
                ],
            };
        },
    });
    expect(wrapper.element).toMatchSnapshot();

    it("10.h.2.1 Carousel has data", () => {
        expect(typeof Carousel.default.data).toBe("function");
    });

    const buttonPrev = wrapper.find('.carousel-control-prev');
    const buttonNext = wrapper.find('.carousel-control-next');

    it("10.h.2.3 next button click", async () => {
        await buttonNext.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.2.4 next button one more click", async () => {
        await buttonNext.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.2.5 next button one more click", async () => {
        await buttonNext.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.2.6 prev button click", async () => {
        await buttonPrev.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.2.7 prev button one more click", async () => {
        await buttonPrev.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.2.8 next button click", async () => {
        await buttonNext.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });
});
