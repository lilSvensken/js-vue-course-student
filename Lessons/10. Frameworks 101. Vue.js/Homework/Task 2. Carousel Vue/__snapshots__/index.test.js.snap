// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.3 next button click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.4 next button one more click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.5 next button one more click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.6 prev button click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.7 prev button one more click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;

exports[`10.h.2 Carousel 10.h.2.8 next button click 1`] = `
<div
  class="carousel"
>
  <div
    class="carousel-inner"
  >
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-1.jpg"
      />
    </div>
    <div
      class="carousel-item"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-2.jpg"
      />
    </div>
    <div
      class="carousel-item active"
    >
      <img
        class="d-block w-100"
        src="/Lessons/09. Frameworks 101. Vue.js/Homework/Task 2. Carousel Vue/assets/carousel-3.jpg"
      />
    </div>
  </div>
   
  <button
    class="carousel-control-prev"
  >
    <span
      class="carousel-control-prev-icon"
    />
  </button>
   
  <button
    class="carousel-control-next"
  >
    <span
      class="carousel-control-next-icon"
    />
  </button>
</div>
`;
